# HG changeset patch
# User Alexander Mamonov <mamonov@gmail.com>
# Date 1245856393 14400
#      Wed Jun 24 11:13:13 2009 -0400
# Node ID 5497b0795fb96ce737c21cd7adcbe0fb99aa3abc
# Parent  8bec23396924aaac06ff26cf9963efe0beec7bdc
plot3.m: correctly compute offsets for property/value pairs

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2009-06-24  Alexander Mamonov  <mamonov@gmail.com>
+
+	* plot/plot3.m: Correctly compute offsets for property/value pairs.
+
 2009-06-23  Alexander Mamonov  <mamonov@gmail.com>
 
 	* image/imread.m: New test.
 
 2009-06-22  Ben Abbott <bpabbott@mac.com>
 
 	* plot/grid.m: Add missing semi-colon. Allow grid to be toggled
 	on/off for each axis independently. Gnuplot requires that minor
diff --git a/scripts/plot/plot3.m b/scripts/plot/plot3.m
--- a/scripts/plot/plot3.m
+++ b/scripts/plot/plot3.m
@@ -140,17 +140,17 @@ function retval = plot3 (varargin)
 	  property_set = 1;
 	  continue;
 	else
 	  fmt_set = 1;
 	  while (arg < nargin && ischar (varargin{arg+1}))
 	    if (nargin - arg < 2)
 	      error ("plot3: properties must appear followed by a value");
 	    endif
-	    properties = [properties, varargin(arg:arg+1)];
+	    properties = [properties, varargin(arg+1:arg+2)];
 	    arg += 2;
 	  endwhile
 	endif
       else
 	properties = [properties, new_cell];
 	property_set = 1;
 	continue;
       endif
